src/Analyzer.Core/TemplateAnalyzer.cs

```diff
@@ -131,28 +174,65 @@ public IEnumerable<IEvaluation> AnalyzeTemplate(string template, string paramete
                     this.logger?.LogDebug("Running PowerShell rule engine");
                     evaluations = evaluations.Concat(this.powerShellRuleEngine.AnalyzeTemplate(templateContext));
                 }
+               
+                // Recursively handle nested templates 
+                dynamic jsonTemplate = JsonConvert.DeserializeObject(template);
+                dynamic processedTemplateResources = templatejObject["resources"];
+                var processedTemplateResourcesWithLineNumbers = templateContext.OriginalTemplate["resources"];
 
-                // For each rule we don't want to report the same line more than once
-                // This is a temporal fix
-                var evalsToValidate = new List<IEvaluation>();
-                var evalsToNotValidate = new List<IEvaluation>();
-                foreach (var eval in evaluations)
+                for (int i = 0; i < processedTemplateResources.Count; i++)
                 {
-                    if (!eval.Passed && eval.Result != null)
-                    {
-                        evalsToValidate.Add(eval);
-                    }
-                    else
+                    var currentProcessedResource = processedTemplateResources[i];
+                    dynamic currentProcessedResourceWithLineNumbers = processedTemplateResourcesWithLineNumbers[i];
+
+                    if (currentProcessedResource.type == "Microsoft.Resources/deployments")
                     {
-                        evalsToNotValidate.Add(eval);
+                        var nestedTemplateWithLineNumbers = currentProcessedResourceWithLineNumbers.properties.template;
+                        dynamic populatedNestedTemplate = JsonConvert.DeserializeObject(JsonConvert.SerializeObject(nestedTemplateWithLineNumbers));
+                        // Get the offset
+                        int nextLineNumberOffset = (nestedTemplateWithLineNumbers as IJsonLineInfo).LineNumber + lineNumberOffset - 1; // off by one
+                        // Check whether scope is set to inner or outer
+                        var scope = currentProcessedResourceWithLineNumbers.properties.expressionEvaluationOptions?.scope;
```

Comment:
Use `.InsensitiveToken()`

Reply:
where am I supposed to use InsensitiveToken here exactly?

Reply:
Once you correct variables to be JToken (or var) instead of dynamic, it will look something like `var scope = currentProcessedResourceWithLineNumbers.InsensitiveToken("properties.expressionEvaluationOptions.scope");`